src/components/world.js

```diff
@@ -42,6 +51,94 @@ export default class World extends Component {
     this.lastTime = currTime;
   };
 
+  onInit = (...args) => {
+    if (this.props.debug) {
+      this.setupDebugRenderer();
+    }
+
+    this.props.onInit(...args);
+  };
+
+  stopDebugRendering = () => {
+    if (this._render) {
+      Render.stop(this._render);
+      delete this._render;
+    }
+  };
+
+  getDebugProps = () => {
+    const debugProps = Object.assign(
+      {
+        offset: {},
+        // transparent background to see sprites, etc, in the world
+        background: 'rgba(0, 0, 0, 0)',
+      },
+      this.props.debug || {}
+    );
+
+    debugProps.offset = Object.assign(
+      {
+        x: 0,
+        y: 0,
+      },
+      debugProps.offset
+    );
+
+    return debugProps;
+  };
+
+  setupDebugRenderer = () => {
+
+    if (!this._debugRenderElement) {
+      return;
+    }
+
+    const { renderWidth, renderHeight, scale } = this.context;
+    const { offset, background } = this.getDebugProps();
+
+    const width = renderWidth / scale;
+    const height = renderHeight / scale;
+
+    this._render = Render.create({
+      canvas: this._debugRenderElement,
+      // Auto-zoom the canvas to the correct game area
+      bounds: {
+        min: {
+          x: offset.x,
+          y: offset.y,
+        },
+        max: {
+          x: offset.x + width,
+          y: offset.y + height,
+        },
+      },
+      options: {
+        wireframeBackground: background,
+        width: renderWidth,
+        height: renderHeight,
+      },
+    });
+
+    // Setting this as part of `.create` crashes Chrome during a deep clone. :/
+    // My guess: a circular reference
+    this._render.engine = this.engine;
```

Comment:
This was a head scratcher for sure! Other than fixing the implementation inside `matter-js`, I couldn't find any other way to stop Chrome from crashing during the deep merge above.